lms/djangoapps/ytp_api/serializers.py

```diff
@@ -0,0 +1,108 @@
+# coding=utf-8
+"""
+Serializer for validate data.
+"""
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from rest_framework.serializers import HyperlinkedModelSerializer
+
+from openedx.core.djangoapps.user_api.accounts.api import check_account_exists
+from openedx.core.djangoapps.user_api.serializers import ReadOnlyFieldsSerializerMixin
+from student.models import CourseEnrollment, LanguageProficiency, UserProfile
+
+
+class UserSerializer(HyperlinkedModelSerializer):
+    """
+    Class that serializes the information of User model.
+
+    Update and validate User information.
+    """
+
+    class Meta(object):
+        model = User
+        read_only_fields = ('password', 'username')
+
+    def update(self, instance, data):
+        """
+        Update User instance by the data dictionary.
+
+        Method updates User instance if data is correct.
+        :param instance: User instance.
+        :param data: Dictionary with user's data.
+        :return:
+        """
+        if not isinstance(instance, User):
+            raise ValidationError("The instance must be the User type.")
+
+        if isinstance(data,dict):
+            username = data.get("username")
+            if username and (instance.username != username):
+                instance.username = self._validate_username(data)
+            email = data.get("email")
+            if email and (instance.email != email):
+                self._check_email_unique(data.get("email"))
```

Comment:
data.get('email') already set to the email variable I would use it instead of getting this value again.